cdap-watchdog/src/main/java/io/cdap/cdap/logging/appender/CompositeLogAppender.java

```
@@ -0,0 +1,120 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A log appender that delegates logging events to a list of other log appenders.
+ */
+public class CompositeLogAppender extends LogAppender {
+
+  private static final org.slf4j.Logger LOG = LoggerFactory.getLogger(CompositeLogAppender.class);
+  private final List<LogAppender> appenders;
+
+  /**
+   * Constructs a CompositeLogAppender with the given list of appenders.
+   *
+   * @param appenders the list of {@link LogAppender} instances to delegate logging events to
+   */
+  public CompositeLogAppender(List<LogAppender> appenders) {
+    if (appenders == null) {
+      throw new IllegalArgumentException("Appenders list cannot be null");
+    }
+
+    // Make the appenders list unmodifiable to ensure thread safety and prevent external
+    // modifications.
+    this.appenders = Collections.unmodifiableList(new ArrayList<>(appenders));
+    setName(getClass().getName());
+  }
+
+  @Override
+  public void start() {
+    appenders.forEach(appender -> safelyExecute(
+        appender::start,
+        String.format("Failed to start appender: %s", appender.getName())
+    ));
+    super.start();
+  }
+
+  @Override
+  public void stop() {
+    super.stop();
+    appenders.forEach(appender -> safelyExecute(
+        appender::stop,
+        String.format("Failed to stop appender: %s", appender.getName())
+    ));
+  }
+
+  @Override
+  protected void appendEvent(LogMessage logMessage) {
+    appenders.forEach(appender -> safelyExecute(
+        () -> appender.appendEvent(logMessage),
+        String.format("Failed to append log message to appender: %s. Message: %s",
+            appender.getName(), logMessage.getFormattedMessage())
+    ));
+  }
+
+  @Override
+  public void setContext(Context context) {
+    super.setContext(context);
+    appenders.forEach(appender -> safelyExecute(
```

Is it ok to ignore the errors using `safelyExecute()` in case of `setContext()` method?

If errors are ignored, it can lead to undesirable state.

---

removed this safelyExecute function now we are combining the errors and throwing that